bin/pygrb/pycbc_pygrb_efficiency

```diff
@@ -0,0 +1,672 @@
+    found_trigs = ppu.load_injections(found_file, vetoes)
+    found_injs = ppu.load_injections(found_file, vetoes, sim_table=True)
+
+    logging.info("Missed/found injections/triggers loaded.")
+
+    # Extract columns of found injections
+    found_inj = {}
+    found_inj['ra'] = np.asarray(found_injs.get_column('longitude'))
+    found_inj['dec'] = np.asarray(found_injs.get_column('latitude'))
+    found_inj['time'] = np.asarray(found_injs.get_column('geocent_end_time')) +\
+                        np.asarray(found_injs.get_column('geocent_end_time_ns')*\
+                                   10**-9)
+    found_inj['dist'] = np.asarray(found_injs.get_column('distance'))
+
+    # Extract columns of triggers
+    found_trig = {}
+    found_trig['mchirp'] = np.asarray(found_trigs.get_column('mchirp'))
+    found_trig['bestnr'] = ppu.get_bestnrs(found_trigs, q=chisq_index,
+                                           n=chisq_nhigh,
+                                           null_thresh=null_thresh,
+                                           snr_threshold=snr_thresh,
+                                           sngl_snr_threshold=sngl_snr_thresh,
+                                           chisq_threshold=new_snr_thresh,
+                                           null_grad_thresh=null_grad_thresh,
+                                           null_grad_val=null_grad_val)
+
+    # Construct conditions for injection:
+    # 1) found louder than background,
+    zero_fap = np.zeros(len(found_injs)).astype(np.bool)
+    zero_fap_cut = found_trig['bestnr'] > max_bestnr
+    zero_fap = zero_fap | (zero_fap_cut)
+
+    # 2) found (bestnr > 0) but not louder than background (non-zero FAP)
+    nonzero_fap = ~zero_fap & (found_trig['bestnr'] != 0)
+
+    # 3) missed after being recovered (i.e., vetoed) are not used here
+    # missed = (~zero_fap) & (~nonzero_fap)
+
+    # Non-zero FAP triggers (g_ifar)
+    g_ifar = {}
+    g_ifar['bestnr'] = found_trig['bestnr'][nonzero_fap]
+    g_ifar['stat'] = np.zeros([len(g_ifar['bestnr'])])
+    for ix, (mc, bestnr) in \
+                enumerate(zip(found_trig['mchirp'][nonzero_fap], g_ifar['bestnr'])):
+        g_ifar['stat'][ix] = (full_time_veto_max_bestnr > bestnr).sum()
+    g_ifar['stat'] = g_ifar['stat'] / total_trials
+
+    # Set the sigma values
+    inj_sigma = found_trigs.get_sigmasqs()
+    # If the sigmasqs are not populated, we can still do calibration errors,
+    # but only in the 1-detector case
+    for ifo in ifos:
+        if sum(inj_sigma[ifo] == 0):
+            logging.info("%s: sigmasq not set for at least one trigger.", ifo)
+        if sum(inj_sigma[ifo] != 0) == 0:
+            logging.info("%s: sigmasq not set for any trigger.", ifo)
+            if len(ifos) == 1:
+                msg = "This is a single ifo analysis. "
+                msg += "Setting sigmasq to unity for all triggers."
+                logging.info(msg)
+                inj_sigma[ifo][:] = 1.
```

Comment:
Looks like this block can stay outside of the `for` loop?

Reply:
But, we do not want to set `inj_sigma[ifo][:] = 1.` unless `sum(inj_sigma[ifo] != 0) == 0` (sigma-squared values are not provided for any trigger) and `len(ifos) == 1` (we are dealing with a single ifo).  By moving this block outside the `for`, we would set `inj_sigma[ifo][:] = 1.` every time we have a single ifo analysis.

Reply:
Ok I see, I just misinterpreted the intent. Thanks.